libs/stats/odc/stats/_fc_percentiles.py

```diff
@@ -0,0 +1,110 @@
+from odc.algo.io import load_with_native_transform
+from odc.algo import keep_good_only
+from odc.algo._percentile import xr_percentile
+from odc.algo._masking import _nodata_fuser, _or_fuser
+from .model import StatsPluginInterface
+from . import _plugins
+
+
+class StatsFCP(StatsPluginInterface):
+    
+    NAME = "ga_fc_percentiles"
+    SHORT_NAME = NAME
+    VERSION = "0.0.1"
+    PRODUCT_FAMILY = "fc_percentiles"
+
+    def __init__(
+        self,
+        resampling: str = "bilinear",
+    ):
+        self.resampling = resampling
+
+    @property
+    def measurements(self) -> Tuple[str, ...]:
+        _measurments = [f"{b}_pc_{p}" for b, p in product(["pv", "bs", "npv", "ue"], ["10", "50", "90"])]
+        _measurments.append("is_ever_wet")
+        return _measurments
+
+    @staticmethod
+    def _native_tr(xx):
+        """
+        Loads in the data in the native projection. It performs the following:
+
+        1. Loads all the fc and WOfS bands
+        2. Set high slope terrain flag to 0
+        3. Extracts the clear dry flags from WOfS
+        4. Drops the WOfS band
+        5. Masks out all pixels that are not clear and dry to a nodata value of 255
+        6. Discards the clear dry flags
+        """
+
+        # set terrain flag to zero
+        water = da.bitwise_and(xx["water"], 0b11101111)
+        xx = xx.drop_vars(["water"])
+
+        # use the dry flag to indicate good measurements
+        dry = water == 0
+        xx = keep_good_only(xx, dry, nodata=255)
+        return xx
+
+    @staticmethod
+    def _native_tr_wet(xx):
+        """
+        Loads in the data in the native projection. It performs the following:
+
+        1. Loads the WOfS band
+        2. Set high slope terrain flag to 0
+        3. Extracts clear wet flags from WOfS
+        """
+
+        # set terrain flag to zero
+        water = da.bitwise_and(xx["water"], 0b11101111)
+        wet = water == 128
+        return wet
+
+    def input_data(self, task: Task) -> xr.Dataset:
+
+        chunks = {"y": -1, "x": -1}
+
+        xx = load_with_native_transform(
+            task.datasets,
+            bands=["water", "pv", "bs", "npv", "ue"],
+            geobox=task.geobox,
+            native_transform=self._native_tr,
+            fuser=_nodata_fuser,
+            groupby="solar_day",
+            resampling=self.resampling,
+            chunks=chunks,
+        )
+
+        xx["wet"] = load_with_native_transform(
+            task.datasets,
+            bands=["water"],
+            geobox=task.geobox,
+            native_transform=self._native_tr_wet,
+            fuser=_or_fuser,
+            groupby="solar_day",
+            resampling=self.resampling,
+            chunks=chunks,
+        )
+
+        return xx
+
+    def reduce(self, xx: xr.Dataset) -> xr.Dataset:        
+        yy = xr_percentile(xx.drop_vars(["wet"]), [0.1, 0.5, 0.9], nodata=255)
+        yy["is_ever_wet"] = _or_fuser(xx["wet"]).squeeze(xx["wet"].dims[0], drop=True)
```

Comment:
I don't think `is_ever_wet` is what we need on the output. But rather an enumerated type with value like
- valid (at least 3 valid dry obs)
- none (no obs at all)
- not sure ([1-3] obs and all dry)
- sure not ([1-3] obs some wet)

Reply:
I thought the `is_ever_wet` band could be combined with the `fc_percentiles` bands to derive this info:

- <= `100` data value -> valid
- `255` data value + `is_ever_wet=False` -> not sure
- `255` data value + `is_ever_wet=True` -> sure not

But I can add this explicitly as an enum type 👍